solver/jobs.go

```diff
@@ -872,6 +872,9 @@ func (s *sharedOp) Exec(ctx context.Context, inputs []Result) (outputs []Result,
 		return nil, nil, err
 	}
 	r := res.(*execRes)
+	if r == nil {
```

Comment:
This looks correct but is quite a weird construction. Normally typed nil should be avoided and `res=nil` should be checked. Maybe a comment saying "callback to `s.g.Do` guarantees that res is always typed `*execRes` if `err`  or `res` is nil" would help.

Reply:
Yeah, it's a weird construction. I've reworked the code so that we can just check `res==nil`.

To do that, I've changed how the method above handles returns - returning `s.execRes` when `s.execRes == nil` is [*not* the same as just returning `nil`](https://stackoverflow.com/questions/62441652/how-do-nil-interface-values-work-in-golang#62441926), and the new code distinguishes between these two cases.